site/content/3.12/aql/high-level-operations/upsert.md

```diff
@@ -63,6 +66,50 @@ UPDATE { logins: OLD.logins + 1 } IN users
 Note that in the `UPDATE` case it is possible to refer to the previous version of the
 document using the `OLD` pseudo-value.
 
+## Filter matching syntax
+
+The alternative syntax for `UPSERT` operations allows you to use more flexible
+filter conditions to look up documents. This means that the expression you use
+to look up a document does not have to be an object literal.
+
+<pre><code>UPSERT FILTER<em>filter-condition</em>
+INSERT <em>insertExpression</em>
+UPDATE <em>updateExpression</em>
+IN <em>collection</em></code></pre>
+
+<pre><code>UPSERT FILTER<em>filter-condition</em>
+INSERT <em>insertExpression</em>
+REPLACE <em>updateExpression</em>
+IN <em>collection</em></code></pre>
+
+The filter condition for the lookup can make use of the pseudo-variable
+`CURRENT` to access the lookup document and apply more filters on it than just
+equality matches. Example:
```

Comment:
The "more than equality matches" part is followed by a  equality marching example. Would be better to move that statement down, and perhaps add another example using >= or something like this. Here, we could state whether all FILTER conditions are allowed.